solvebio/resource/object.py

```diff
@@ -277,6 +278,43 @@ def get_or_create_by_full_path(cls, full_path, **kwargs):
                              client=_client,
                              **kwargs)
 
+    def _archive(self, archive_folder):
+        from solvebio.cli.data import _create_folder
+
+        # Create timestamped archive filename
+        date_format = "%Y-%m-%d_%Hh%Mm%Ss_%Z"
+        timestamp = datetime.now().strftime(date_format)
+        file_extension = self.filename.split(".")[-1]
```

Comment:
What about compressed files? .json.gz, etc...

Reply:
Updated to match logic in API for filename incrementing.